core/src/test/resources/sql/sub-query.iq

```diff
@@ -3257,6 +3257,34 @@ EnumerableCalc(expr#0..7=[{inputs}], expr#8=[null:BOOLEAN], proj#0..8=[{exprs}])
   EnumerableTableScan(table=[[scott, EMP]])
 !plan
 
+# [CALCITE-4913] Correlated variables in a select list are not deduplicated
+select e.deptno,
+       (select count(*) from emp where e.deptno > 0) as c1,
+       (select count(*) from emp where e.deptno > 0 and e.deptno < 10) as c2
+  from emp e;
+
++--------+----+----+
+| DEPTNO | C1 | C2 |
++--------+----+----+
+|     10 | 14 |  0 |
+|     10 | 14 |  0 |
+|     10 | 14 |  0 |
+|     20 | 14 |  0 |
+|     20 | 14 |  0 |
+|     20 | 14 |  0 |
+|     20 | 14 |  0 |
+|     20 | 14 |  0 |
+|     30 | 14 |  0 |
+|     30 | 14 |  0 |
+|     30 | 14 |  0 |
+|     30 | 14 |  0 |
+|     30 | 14 |  0 |
+|     30 | 14 |  0 |
++--------+----+----+
+(14 rows)
+
+!ok
```

Comment:
Apart from the result please also include the `!plan`

Reply:
fixed